lib/components/scanner/scanner_view.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+import 'package:seeds/components/scanner/components/qr_code_view.dart';
+import 'package:seeds/components/scanner/interactor/viewmodels/scanner_bloc.dart';
+import 'package:seeds/utils/string_extension.dart';
+
+/// Scanner SCREEN
+class ScannerView extends StatefulWidget {
+  final ScannerBloc _scannerBloc = ScannerBloc();
+  final ValueSetter<String> onCodeScanned;
+
+  ScannerView({Key? key, required this.onCodeScanned}) : super(key: key);
+
+  void scan() => _scannerBloc.add(const Scan());
+
+  @override
+  _ScannerViewState createState() => _ScannerViewState();
+}
+
+class _ScannerViewState extends State<ScannerView> {
+  final GlobalKey _qrKey = GlobalKey(debugLabel: 'QR');
```

Comment:
I don't think this global key be necessary too. I let a issue in the repo to know if this is a requirement.
https: //github.com /juliuscanute /qr_code_scanner /issues/ 472

Reply:
yeah we need it its a requirement for iOS.
https://github.com/juliuscanute/qr_code_scanner/issues/226